### PR TITLE
Fix: check on justfile

### PR DESCRIPTION
**Pull request by jaoleal**
now check in justfile evoke compile-all.sh

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ watch:
...
     typst w slides.typ
 
-# Compile and Run the C code
-c:
+# Compile the C code in "code/"
+check:
     @echo "Compiling and Running the C code"
-    @echo "TODO!"
\ No newline at end of file
+    ./compile-all.sh
```
Comment (storopoli) [inline justfile:21]: Change to
```
check:
    #!/usr/bin/env bash
    echo "compiling all *.c code to output"
    rm -r ./output;mkdir ./output
    for FILE in $(ls ./solutions/)
    do
	    cc ./solutions/$FILE -o ./output/$FILE.out
    done
```

Changes:

- Using `/usr/bin/env bash` is more failsafe than `/bin/bash`
- `cc` is more failsafe than `clang`

**Review by storopoli — CHANGES_REQUESTED** [review]
There's a bunch of ~crap~ artifacts like the whole `output/` dir

File: justfile (hunk 1 above)
Comment (jaoleal) [reply justfile:21]: Okay

**Comment by jaoleal** [comment]
> There's a bunch of ~crap~ artifacts like the whole `output/` dir

Its where the compiled things go, wanna change ?

**Comment by jaoleal** [comment]
Done and rebased. Wonder if the "check" thing is sounds more like "Check if my code is correct"; if it is so, we could do a python script that matches the output of the solution and the code done by the students. Otherwise "check if my c code is compiling"

**Comment by storopoli** [comment]
> Done and rebased. Wonder if the "check" thing is sounds more like "Check if my code is correct"; if it is so, we could do a python script that matches the output of the solution and the code done by the students. Otherwise "check if my c code is compiling"

No python please...

**Comment by jaoleal** [comment]
>No python please...
it's his only purpose, scripting...

File: justfile [hunk 2]
```diff
@@ -15,7 +15,12 @@ watch:
...
\ No newline at end of file
+# Compile the C code in "code/"
+check:
+    #!/usr/bin/env bash
+    echo "compiling all *.c code to output"
+    rm -r ./output;mkdir ./output
+    for FILE in $(ls ./solutions/)
+    do
+	    cc ./solutions/$FILE -o ./output/$FILE.out
+    done
```
Comment (storopoli) [inline justfile:26]: This is not a `check` anymore since it compiles the `solutions` directory and not the `code` directory.

File: justfile [hunk 3]
```diff
@@ -15,7 +15,12 @@ watch:
...
 
-# Compile and Run the C code
-c:
-    @echo "Compiling and Running the C code"
-    @echo "TODO!"
\ No newline at end of file
+# Compile the C code in "code/"
+check:
+    #!/usr/bin/env bash
+    echo "compiling all *.c code to output"
```
Comment (storopoli) [inline justfile:21]: I prefer to use the `just` builtin `@echo` instead of bash's `echo`
```suggestion
    @echo "compiling all *.c code to output"
```

**Review by storopoli — CHANGES_REQUESTED** [review]

**Comment by storopoli** [comment]
> > No python please...
> 
> it's his only purpose, scripting...

Yeah tell this to a first-year undergraduate using the repo...

File: justfile [hunk 4]
```diff
@@ -15,7 +15,12 @@ watch:
     @echo "Watching slides"
     typst w slides.typ
 
-# Compile and Run the C code
-c:
-    @echo "Compiling and Running the C code"
-    @echo "TODO!"
\ No newline at end of file
+# Compile the C code in "code/"
```
Comment (storopoli) [inline justfile:18]: This docstring needs updating:

```suggestion
# Compile the C code in "solutions/"
```

**Review by storopoli — CHANGES_REQUESTED** [review]

File: justfile (hunk 4 above)
Comment (jaoleal) [reply justfile:18]: inst it right ? the solution dir contains the corrected code.

Comment (storopoli) [reply justfile:18]: Yes but look down below you are doing:

```just
    for FILE in $(ls ./solutions/)
    do
	    cc ./solutions/$FILE -o ./output/$FILE.out
    done
```

**Comment by jaoleal** [comment]
Did the requested changes, looks like that just`s built in commands doesnt work well while using shebangs...

**Pull request merged**